### PR TITLE
radosgw-admin: skip GC init on read-only admin ops

### PR DESCRIPTION
**Pull request by mkogan1**
Fixes: https://tracker.ceph.com/issues/50520

Signed-off-by: Mark Kogan <mkogan@redhat.com>

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by mkogan1** [comment]
jenkins test make check

File: src/rgw/rgw_rados.h [hunk 1]
```diff
@@ -480,6 +480,7 @@ class RGWRados
   RGWIndexCompletionManager *index_completion_manager{nullptr};
 
   bool use_cache{false};
+  bool readonly{false};
```
Comment (mattbenjamin) [inline src/rgw/rgw_rados.h:483]: I guess this makes sense.  I am more strongly preferring the idea of pulling admin into a remote service all the time.

**Review by mattbenjamin — APPROVED** [review]
overall, looks fine to me

**Review by dang — APPROVED** [review]

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by mkogan1** [comment]
updated as discussed on stadup,
teuthology run run of the `rgw` suite:
https://pulpito.ceph.com/mkogan-2021-04-27_14:46:40-rgw-wip-rgw-gc-max-objsf2_i04-distro-basic-smithi/
none of the failures related  to `radosgw-admin`

**Review by mattbenjamin — APPROVED** [review]
this actually now does look good to me, read it earlier in the day ;)

**Pull request merged**